content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md

```diff
@@ -0,0 +1,1711 @@
+
+For more information about the service LoadBalancer component you can refer to the [Kubernetes official documentation](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer).
+
+<br />
+
+</Tabs.TabPane>
+<Tabs.TabPane tab="Terraform" key="terraform-application">
+
+
+### Add the Manifest
+
+Open the folder where you have the profile configuration
+```bash
+$ cd terraform-profile
+```
+
+Edit the file *cluster_profile.tf* and add the manifest pack inside the profile definition
+
+```terraform
+resource "spectrocloud_cluster_profile" "profile" {
+  ...
+  pack {
+    name = "hello-universe"
+    type = "manifest"
+    tag  = "1.0.0"
+    manifest {
+      name    = "hello-universe"
+      content = file("manifest.yaml")
+    }
+  }
+}
+```
+
+This new manifest pack contains the configurations of the application through the file *manifest.yaml*. So, create the file *manifest.yaml* with the Kubernetes configuration to deploy the app:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: hello-universe-service
+spec:
+  type: LoadBalancer
+  selector:
+    app: hello-universe
+  ports:
+  - protocol: TCP
+    port: 8080
+    targetPort: 8080
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hello-universe-deployment
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: hello-universe
+  template:
+    metadata:
+      labels:
+        app: hello-universe
+    spec:
+      containers:
+      - name: hello-universe
+        image: ghcr.io/spectrocloud/hello-universe:1.0.10
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 8080
+```
+
+In this code example, we deploy the [*hello-universe*](https://github.com/spectrocloud/hello-universe) demo application.
+
+We set two replicas for the application to simulate a distributed environment with a redundant web application deployed on Kubernetes. In front of them, we add a load balancer service to route requests across all replica containers as best practice to maximize the workload and to expose a single access point to the web application.
+
+For more information about the service LoadBalancer component you can refer to the [Kubernetes official documentation](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer).
+
+<br />
+
+Then, use Terraform to push the modification to Palette and finalize the application deployment.
+
+So, first, check the validation of the configuration files:
+```bash
+$ terraform validate
+```
+
+Create the execution plan with the additional modifications to the already existant configuration:
+```bash
+$ terraform plan
+```
+
+Finally, apply the modifications there are in the plan to execute them and create the infrastructure:
+```bash
+$ terraform apply
+```
+
+</Tabs.TabPane>
+</Tabs>
+
+<br />
+
+
+### Deploy
```

Comment:
This step is not needed with Terraform. If you keep everything in the same folder, the cluster will receive the new update. This problem will go away when you simplify everything under two main tabs UI workflow and Terraform.

Reply:
Removed for Terraform